homeassistant/components/alarm_control_panel/manual.py

```diff
@@ -63,19 +94,30 @@ class ManualAlarm(alarm.AlarmControlPanel):
     or disarm if `disarm_after_trigger` is true.
     """
 
-    def __init__(self, hass, name, code, pending_time,
-                 trigger_time, disarm_after_trigger):
+    def __init__(self, hass, name, code, pending_time, trigger_time,
+                 disarm_after_trigger, state_settings):
         """Init the manual alarm panel."""
         self._state = STATE_ALARM_DISARMED
         self._hass = hass
         self._name = name
         self._code = str(code) if code else None
-        self._pending_time = datetime.timedelta(seconds=pending_time)
         self._trigger_time = datetime.timedelta(seconds=trigger_time)
         self._disarm_after_trigger = disarm_after_trigger
         self._pre_trigger_state = self._state
         self._state_ts = None
 
+        self._pending_time_by_state = {}
+        for state in PENDINGABLE_STATES:
```

Comment:
This also looks like it could be done in the config schema.

Reply:
I'll make the change

Reply:
Actually are you talking about the code below? The reason why I'm doing this is because I want to use "pending_time" as the default time if there's no override in the state_setting

Reply:
I'm talking about the whole iteration and `_pending_time_by_state` setting. Break this out into a config schema validator that is passed the whole platform config. Use the validator in a `vol.All` call as last argument, so that the existing validation is done before this new validator is checked. That will make sure that defaults are in place before this validator is run.
```py
PLATFORM_SCHEMA = vol.Schema(vol.All({
    ...
}, state_validator))
```
By passing the whole config to the validator you will have access to pending time and can set that as default for each state. See https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/scene/__init__.py#L26-L53 for another example.

The you can just pass the whole validated config to the entity and assign that to an instance attribute.